redirect/routes.php

```diff
@@ -262,20 +266,34 @@ class Route {
         ],
     ];
 
-    public function getTargetOfSub($subdomain) {
+    public function getTargetOfSub($subdomain, $subsub) {
+
         if ($subdomain === 'json') {
             header('Content-type: application/json');
             die(json_encode($this->routes));
         }
+
         if (isset($this->synonyms[ $subdomain ])) {
             $subdomain = $this->synonyms[ $subdomain ];
         }
 
+        switch ($subsub) {
+            case 'm' :  
+                // This is a moodle redirect like m.info1.tum.sexy
+                $moodle_id = $this->routes[ $subdomain ][ 'moodle_id' ];
+                if (!isset($moodle_id)) {
+                    return $this->routes[ $subdomain ][ 'target' ];  // Fallback to target if moodle id is unknown
+                }
+                return 'https://www.moodle.tum.de/course/view.php?id=' . $moodle_id;
+            default: 
+                break;
```

Comment:
Why have a default case if it does nothing?

Reply:
@jonathangpk

Reply:
I didn't know if it was required in PHP, so I added it 😆

Reply:
Nope, not required. Please show me a language where this is required - I don't know any.

Reply:
Don't know one either ^^
Changed it

Reply:
I can tell you one:
In Swift every switch statement must be exhaustive. So if you don't provide a catch for every possibility, you **need** a default case. :wink:

Reply:
Nobody uses swift. :boom:

Reply:
Nah, I really enjoy Swift. It's just Xcode that sucks 🙈